packages/web/src/features/translation/VerseSelector.tsx

```diff
@@ -0,0 +1,53 @@
+import { KeyboardEvent } from "react";
+import { useTranslation } from "react-i18next";
+import { Icon } from "../../shared/components/Icon";
+import TextInput from "../../shared/components/TextInput";
+import {
+  decrementVerseId,
+  generateReference,
+  incrementVerseId,
+  parseReference,
+  parseVerseId
+} from "./verse-utils";
+
+export interface VerseSelectorProps {
+  verseId: string;
+  onVerseChange: (verseId: string) => any;
+}
+
+export function VerseSelector({ verseId, onVerseChange }: VerseSelectorProps) {
+  const { t } = useTranslation(["translation", "bible"]);
+  const verseInfo = parseVerseId(verseId);
+
+  const onKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
+    if (e.key == "Enter") {
+      const newReference = (e.target as HTMLInputElement).value;
+      (e.target as HTMLInputElement).value = "";
```

Comment:
I had to cast, otherwise I get errors.  
<img width="395" alt="image" src="https://github.com/arrocke/gloss-translation/assets/17209828/b249934e-e588-483f-8ecc-eb1224ab8ee4">